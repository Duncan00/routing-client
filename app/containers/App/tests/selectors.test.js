import {
  selectGlobal,
  makeSelectApi,
  makeSelectRouteToken,
  makeSelectRoute,
  makeSelectError,
  makeSelectLoading,
} from '../selectors';

describe('App/selectors', () => {
  describe('selectGlobal', () => {
    describe('When global state is falsy', () => {
      it('should return the global initial state', () => {
        const mockedState = {
          global: null,
        };
        expect(selectGlobal(mockedState)).toMatchSnapshot();
      });
    });
  });

  describe('makeSelectApi', () => {
    const apiSelector = makeSelectApi();
    it('should select api', () => {
      const api = 'mock_api';
      const mockedState = {
        global: {
          api,
        },
      };
      expect(apiSelector(mockedState)).toEqual(api);
    });
  });

  describe('makeSelectApi', () => {
    const apiSelector = makeSelectApi();
    it('should select api', () => {
      const api = 'mock_api';
      const mockedState = {
        global: {
          api,
        },
      };
      expect(apiSelector(mockedState)).toEqual(api);
    });
  });

  describe('makeSelectRouteToken', () => {
    const routeTokenSelector = makeSelectRouteToken();
    it('should select api', () => {
      const routeToken = 'XXX';
      const mockedState = {
        global: {
          user_data: {
            route_token: routeToken,
          },
        },
      };
      expect(routeTokenSelector(mockedState)).toEqual(routeToken);
    });
  });

  describe('makeSelectRoute', () => {
    const routeSelector = makeSelectRoute();
    it('should select api', () => {
      const route = {};
      const mockedState = {
        global: {
          user_data: {
            route,
          },
        },
      };
      expect(routeSelector(mockedState)).toEqual(route);
    });
  });

  describe('makeSelectLoading', () => {
    const loadingSelector = makeSelectLoading();
    it('should select the loading', () => {
      const loading = false;
      const mockedState = {
        global: {
          loading,
        },
      };
      expect(loadingSelector(mockedState)).toEqual(loading);
    });
  });

  describe('makeSelectError', () => {
    const errorSelector = makeSelectError();
    it('should select the error', () => {
      const error = 404;
      const mockedState = {
        global: {
          error,
        },
      };
      expect(errorSelector(mockedState)).toEqual(error);
    });
  });
});
