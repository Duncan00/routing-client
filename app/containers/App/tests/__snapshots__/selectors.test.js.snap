// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App/selectors selectGlobal When global state is falsy should return the global initial state 1`] = `
Object {
  "api": undefined,
  "error": false,
  "loading": false,
  "user_data": Object {
    "route": false,
    "route_token": "",
  },
}
`;
